File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum=0,max=INT_MIN;
        if(nums.size() == 1) return nums[0];
        
        for(int i=0;i<nums.size();i++){
            sum+=nums[i];
            if(sum > max)
                max=sum;
            if(sum < 0){
                if(sum > max)
                    max=sum;
                sum=0;
            }
            
        }
        return max;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0,maxSum = Integer.MIN_VALUE;
        for(int i=0;i<nums.length;i++){
            sum += nums[i];
            if(sum > maxSum ){
                maxSum = sum;
            }
            if(sum < 0)
                sum = 0;
        }
        return maxSum;
    }
}